codecov_cli/helpers/ci_adapters/github_actions.py

```diff
@@ -66,21 +66,21 @@ def _get_slug(self):
         return os.getenv("GITHUB_REPOSITORY")
 
     def _get_branch(self):
-        branch = os.getenv("GITHUB_HEAD_REF")
-        if branch:
-            return branch
+        def remove_prefix(s, prefix):
+            if s.startswith(prefix):
+                return s[len(prefix) :]
+            return s
 
-        branch_ref = os.getenv("GITHUB_REF")
+        head_ref = os.getenv("GITHUB_HEAD_REF", "")
+        ref = remove_prefix(os.getenv("GITHUB_REF", ""), "refs/heads/")
 
-        if not branch_ref:
-            return None
-
-        match = re.search(r"refs/heads/(.*)", branch_ref)
+        branch = head_ref or ref
 
-        if match is None:
-            return None
+        # branch format for merge queue CI runs:
+        # gh-readonly-queue/<branch-name>/<pr-number>-<pr-name>
+        if branch.startswith("gh-readonly-queue/"):
+            return branch.split("/")[1]
 
-        branch = match.group(1)
         return branch or None
```

Comment:
my worry here is if we depend on `None` to be meaningful versus `''`. Previous logic doesn't account for empty string